radiation/python/radlw/lwrad_gt4py.py

```diff
@@ -0,0 +1,1800 @@
+    elif var == "index" or var == "ia":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_INT)
+    else:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_FLT)
+
+# Initialize local vars for taumol
+for var in locvars_int:
+    if var == "ib":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_2D, DTYPE_INT)
+    elif var == "NGB":
+        locdict_gt4py[var] = create_storage_from_array(
+            np.tile(np.array(ngb)[None, None, :], (npts, 1, 1)),
+            backend,
+            shape_2D,
+            (DTYPE_INT, (ngptlw,)),
+            default_origin=(0, 0),
+        )
+    else:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_INT)
+
+for var in locvars_flt:
+    if var == "taug":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, type_ngptlw)
+    else:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_FLT)
+
+# Initialize local vars for rtrnmc
+for var in locvars_rtrnmc:
+    if var[-3:] == "rad":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, type_nbands)
+    elif var == "fnet" or var == "fnetc" or var == "rfdelp":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_FLT)
+    elif var == "itgas" or var == "ittot":
+        locdict_gt4py[var] = create_storage_zeros(
+            backend, shape_nlp1, (np.int32, (ngptlw,))
+        )
+    elif var == "ib":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_2D, np.int32)
+    else:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, type_ngptlw)
+
+locdict_gt4py["A0"] = create_storage_from_array(a0, backend, shape_nlp1, type_nbands)
+locdict_gt4py["A1"] = create_storage_from_array(a1, backend, shape_nlp1, type_nbands)
+locdict_gt4py["A2"] = create_storage_from_array(a2, backend, shape_nlp1, type_nbands)
+
+# Read in lookup table data for cldprop calculations
+ds = xr.open_dataset("../lookupdata/radlw_cldprlw_data.nc")
+
+absliq1 = ds["absliq1"].data
+absice0 = ds["absice0"].data
+absice1 = ds["absice1"].data
+absice2 = ds["absice2"].data
+absice3 = ds["absice3"].data
+
+lookup_dict = dict()
+lookup_dict["absliq1"] = create_storage_from_array(
+    absliq1, backend, (npts, 1, nlp1 + 1), (np.float64, (58, nbands))
+)
+lookup_dict["absice0"] = create_storage_from_array(
+    absice0, backend, (npts, 1, nlp1 + 1), (np.float64, (2,))
+)
+lookup_dict["absice1"] = create_storage_from_array(
+    absice1, backend, (npts, 1, nlp1 + 1), (np.float64, (2, 5))
+)
+lookup_dict["absice2"] = create_storage_from_array(
+    absice2, backend, (npts, 1, nlp1 + 1), (np.float64, (43, nbands))
+)
+lookup_dict["absice3"] = create_storage_from_array(
+    absice3, backend, (npts, 1, nlp1 + 1), (np.float64, (46, nbands))
+)
+lookup_dict["ipat"] = create_storage_from_array(
+    ipat, backend, (npts, 1, nlp1 + 1), (DTYPE_INT, (nbands,))
+)
```

Comment:
This works and is clear, though you could alternatively directly use ds["absliq1"].data rather than making a separate variable that is used once. You could do a loop over these variables, except they have different sizes specified. One option could be to make a dictionary mapping the names to the unique size, then do a loop over the dictionary like
`cldprop_shapes = {"absliq1": (np.float64, (58, nbands), "absice0": (np.float64, (2,), ...}
for cldprop, cldprop_shape in cldprop_shapes.items():
    lookup_dict[cldprop] = create_storage_from_array(
         ds[cldprop].data, backend, (npts, 1, nlp1 + 1), cldprop_shape)
   )
`
This reduces the repetition and makes it more clear what is unique about each cloud property, and makes the code simpler.
Just a suggestion for a potential style, not a requirement. There are some benefits to this way of spelling out each item, makes it easier to change a single one of them without impacting all the variables.

Reply:
Good idea, I'll change it